doc/api/http.md

```diff
@@ -587,13 +587,21 @@ added: v1.6.0
 -->
 
 * `name` {string}
-* Returns: {string}
+* Returns: {string|string[]}
 
 Reads out a header on the request. Note that the name is case insensitive.
+The type of the return value depends on the arguments provided to
+[`response.setHeader()`][].
 
 Example:
 ```js
-const contentType = request.getHeader('Content-Type');
+request.setHeader('content-type', 'text/html');
+request.setHeader('Content-Length': Buffer.byteLength(body));
+request.setHeader('Set-Cookie', ['type=ninja', 'language=javascript'])
+const contentType = request.getHeader('Content-Type');      // contentType is 'text/html'
```

Comment:
We lint code examples in docs, so these 3 lines will violate `max-len` rule (they exceed 80 characters length). The comments may be moved to their own lines.